widget/textgrid_test.go

```diff
@@ -0,0 +1,134 @@
+package widget
+
+import (
+	"image/color"
+	"testing"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/test"
+	"fyne.io/fyne/theme"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewTextGrid(t *testing.T) {
+	grid := NewTextGridFromString("A")
+	test.WidgetRenderer(grid).Refresh()
+
+	assert.Equal(t, 1, len(grid.Content))
+	assert.Equal(t, 1, len(grid.Content[0]))
+}
+
+func TestTextGrid_LineNumbers(t *testing.T) {
+	grid := NewTextGridFromString("1\n2\n3\n4\n5\n6\n7\n8\n9\n10")
+	grid.ShowLineNumbers = true
+
+	grid.Resize(fyne.NewSize(100, 250))
+	r := test.WidgetRenderer(grid).(*textGridRenderer)
+
+	assert.Equal(t, ' ', rendererCellRune(r, 0, 0))
```

Comment:
This would be cool to be just a comparison to an ASCII string representing the rendered grid.